src/main/java/com/startsteps/ecommerceapi/user/service/EmailServiceImpl.java

```diff
@@ -1,15 +1,24 @@
 package com.startsteps.ecommerceapi.user.service;
 
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
 import org.springframework.mail.SimpleMailMessage;
 import org.springframework.mail.javamail.JavaMailSender;
 import org.springframework.scheduling.annotation.Async;
 import org.springframework.stereotype.Service;
 
 @Service("emailService")
-public class EmailServiceImpl {
-    @Autowired
+@Configuration
+@AllArgsConstructor
+@NoArgsConstructor
+public class EmailServiceImpl implements EmailService{
```

Comment:
is this needed?

Reply:
I need Email Service (java) for sending email for the user if he wants to reset his password. I have the implementation of this method in User Service.

Reply:
I would leave this for last since it is not a part of the project requirement